pydantic/version.py

```diff
@@ -72,8 +77,26 @@ def version_info() -> str:
 
 def check_pydantic_core_version() -> bool:
     """Check that the installed `pydantic-core` dependency is compatible."""
-    # Keep this in sync with the version constraint in the `pyproject.toml` dependencies:
-    return __pydantic_core_version__ == '2.37.2'
+    return __pydantic_core_version__ == _COMPATIBLE_PYDANTIC_CORE_VERSION
+
+
+def _ensure_pydantic_core_version() -> None:  # pragma: no cover
+    if not check_pydantic_core_version():
+        raise_error = True
+        # Do not raise the error if pydantic is installed in editable mode (i.e. in development):
+        if sys.version_info >= (3, 13):  # origin property added in 3.13
```

Comment:
Do we potentially want an env var to silence this for development on older Python versions?

Reply:
I thought about it, but I think most one-time contributors will have the version right on `uv sync`, and for us well I can only recommend using 3.13 or soon 3.14 :smile: